src/pages/index.js

```diff
@@ -78,24 +126,123 @@ addcreateCardButton.addEventListener("click", () => {
   formValidators["add-button-form"].resetValidation();
 });
 
+avatarImage.addEventListener("click", () => {
+  avatarEditModal.open();
+});
+
 // EVENT HANDLERS
 
+function handleLikeCard(cardId) {
+  return api
+    .likeCard(cardId)
+    .then(() => {
+      const card = cardsSection.getItem(cardId);
+      if (card) {
+        card._updateLikeStatus(true);
+      }
+    })
+    .catch((err) => {
+      console.error(`Error liking card: ${err}`);
+    });
+}
+
+function handleUnlikeCard(cardId) {
+  return api
+    .unlikeCard(cardId)
+    .then(() => {
+      const card = cardsSection.getItem(cardId);
+      if (card) {
+        card._updateLikeStatus(false);
+      }
+    })
+    .catch((err) => {
+      console.error(`Error unliking card: ${err}`);
+    });
+}
+
+function handleDeleteButtonClick(cardId) {
+  deleteCardModal.open(cardId);
+}
+
 function handleProfileEditSubmit(inputValues) {
-  userInfo.setUserInfo({
+  profileSaveButton.textContent = "Saving...";
+
+  const updatedUserInfo = {
     name: inputValues.name,
-    job: inputValues.job,
-  });
+    about: inputValues.about,
+  };
 
-  profileModal.close();
+  api
+    .updateUserInfo(updatedUserInfo)
+    .then((updatedData) => {
+      userInfo.setUserInfo(updatedData);
+      profileModal.close();
+    })
+    .catch((err) => {
+      console.error("Error updating user information:", err);
+    })
+    .finally(() => {
+      profileSaveButton.textContent = "Save";
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```